processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -790,4 +790,30 @@ public boolean hasEmptyAccessibleContructor() {
         }
         return hasEmptyAccessibleContructor;
     }
+
+    /**
+     * Searches for the given superclass and collects all type arguments for the given class
+     *
+     * @param superclass the superclass or interface the generic type arguments are searched for
+     * @return a list of type arguments or null, if superclass was not found
+     */
+    public List<Type> determineTypeArguments(Class<?> superclass) {
+        TypeMirror superclassMirror =
+            typeUtils.erasure( elementUtils.getTypeElement( superclass.getCanonicalName() ).asType() );
+        if ( typeUtils.isAssignable( superclassMirror, typeMirror )
```

Comment:
I guess you use `isAssignable` in both directions because you are working with the erasure of `superclass`? I would guess that you could just String-compare the qualified type name instead to save us some CPU cycles (qualifiedName.equals( superclass.getName() )`. That would also eliminate the need to re-create the `TypeElement` / `TypeMirror` instances of `Map` / `Iterable` for each invocation.